internal/cli/commands/new.go

```diff
@@ -0,0 +1,131 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+
+	f "github.com/foundation-go/foundation"
+	"github.com/foundation-go/foundation/internal/cli/helpers"
+	"github.com/foundation-go/foundation/internal/cli/templates"
+)
+
+type newInput struct {
+	FoundationVersion string
+	Name              string
+}
+
+var (
+	newAppFiles = []string{
+		"README.md",
+		".gitignore",
+		"foundation.toml",
+	}
+
+	newServiceFiles = []string{
+		".env.example",
+		"README.md",
+		"cmd/grpc/main.go",
+	}
+)
+
+var New = &cobra.Command{
+	Use:     "new",
+	Aliases: []string{"n"},
+	Short:   "Create a new Foundation application or service",
+	Run: func(cmd *cobra.Command, _ []string) {
+		serviceFlag, err := cmd.Flags().GetBool("service")
+		if err != nil {
+			log.Fatal().Err(err).Msg("Failed to get `--service` flag")
+		}
+
+		input := &newInput{
+			FoundationVersion: f.Version,
+			Name:              cmd.Flag("name").Value.String(),
+		}
+
+		if serviceFlag {
+			newService(input)
+		} else {
+			newApplication(input)
+		}
+	},
+}
+
+func newApplication(input *newInput) {
+	newEntity(input, "application", newAppFiles)
+
+	if helpers.InGitRepository() {
+		log.Info().Msg("Git repository already exists, skipping initialization")
+	} else if err := helpers.RunCommand(input.Name, "git", "init"); err != nil {
+		log.Fatal().Err(err).Msg("Failed to initialize Git repository")
+	}
+}
+
+func newService(input *newInput) {
+	appRoot := helpers.GetApplicationRoot()
+
+	if err := os.Chdir(appRoot); err != nil {
+		log.Fatal().Err(err).Msg("Failed to change directory to application root")
+	}
+
+	newEntity(input, "service", newServiceFiles)
+
+	if err := helpers.RunCommand(input.Name, "go", "mod", "init"); err != nil {
```

Comment:
I have tried it locally and got an error `go: cannot determine module path for source directory /www/one/brand/worker (outside GOPATH, module path must be specified)`.
I think we need to add mod name, for example
```suggestion
	if err := helpers.RunCommand(input.Name, "go", "mod", "init", input.Name); err != nil {
```

Reply:
> outside GOPATH

**Inside** the GOPATH everything works perfectly :)

Reply:
But most users will run it somewhere outside?

Reply:
I was today years old when I learned that living outside of `GOPATH` is an actual thing now.

Your point is absolutely correct, it definitely requires to be addressed.